fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -902,6 +916,109 @@ private static void checkSearchParameterRestrictions(String parameterCode, Searc
+
+        List<Set<String>> spCombinations = null;
+
+        // Retrieve the "resources" config property group.
+        PropertyGroup rsrcsGroup = FHIRConfigHelper.getPropertyGroup(FHIRConfiguration.PROPERTY_RESOURCES);
+        if (rsrcsGroup != null) {
+            List<PropertyEntry> rsrcsEntries = rsrcsGroup.getProperties();
+            if (rsrcsEntries != null && !rsrcsEntries.isEmpty()) {
+                List<String> combinations = null;
+
+                // Try to find search parameter combinations property for matching resource type
+                for (PropertyEntry rsrcsEntry : rsrcsEntries) {
+                    if (resourceType.equals(rsrcsEntry.getName())) {
+                        PropertyGroup resourceTypeGroup = (PropertyGroup) rsrcsEntry.getValue();
+                        if (resourceTypeGroup != null) {
+                            combinations = resourceTypeGroup.getStringListProperty(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_PARAMETER_COMBINATIONS);
+                            break;
+                        }
+                    }
+                }
+
+                // Otherwise, try to find search parameter combinations property for "Resource" resource type
+                if (combinations == null) {
+                    for (PropertyEntry rsrcsEntry : rsrcsEntries) {
+
+                        // Check if matching resource type
+                        if (SearchConstants.RESOURCE_RESOURCE.equals(rsrcsEntry.getName())) {
+                            PropertyGroup resourceTypeGroup = (PropertyGroup) rsrcsEntry.getValue();
+                            if (resourceTypeGroup != null) {
+                                combinations =
+                                        resourceTypeGroup.getStringListProperty(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_PARAMETER_COMBINATIONS);
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                // Convert the delimited combinations to a list of sets
+                if (combinations != null) {
+                    spCombinations = new ArrayList<>();
+                    for (String combination : combinations) {
+                        Set<String> combinationSet = new HashSet<>();
+                        if (!combination.isEmpty()) {
+                            // If any search parameter combination is allowed, return null
+                            if (SEARCH_PARAM_COMBINATION_ANY.equals(combination)) {
+                                return null;
+                            }
+                            for (String spString : combination.split(SEARCH_PARAM_COMBINATION_DELIMITER)) {
+                                if (spString.trim().isEmpty()) {
+                                    throw SearchExceptionUtil.buildNewIllegalStateException();
+                                }
+                                combinationSet.add(spString);
```

Comment:
I think we should be proactive here, and only accept the trimmed codes/combinations.

Reply:
Updated.